### PR TITLE
Fix MaskFormerImageProcessor.post_process_instance_segmentation

### PR DESCRIPTION
**Pull request by alaradirik**
# What does this PR do?
Fixes the `post_process_instance_segmentation` method of `MaskFormerImageProcessor`. This issue mainly affects Mask2Former as it uses MaskFormerImageProcessor and there aren't any MaskFormer models trained on instance segmentation datasets. 

Unlike panoptic segmentation post-processing, the final score of each binary mask proposal is calculated by multiplying the mask proposal score with the class score. `mask_threshold` and `overlap_mask_area_threshold` arguments are not needed anymore, I can either add a warning to deprecate them or leave it as it is for now.

Post-processed results of the `mask2former-swin-small-coco-instance` model inference:
download

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [ X] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/main/CONTRIBUTING.md#start-contributing-pull-requests),
      Pull Request section?
- [x] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.
- [ ] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/main/docs#writing-source-documentation).
- [ ] Did you write any new necessary tests?

### CODE REVIEW

**Review by sgugger — APPROVED** [review]
Thanks for the fix!

**Review by NielsRogge — APPROVED** [review]
Thanks for fixing! Should we add a corresponding test for it, which verifies the postprocessed results?

**Comment by alaradirik** [comment]
> Thanks for fixing! Should we add a corresponding test for it, which verifies the postprocessed results?

I added a test but Mask2Former, unlike MaskFormer, outputs segmentation maps of shape (96, 96) instead of the preprocessed input size for efficiency. They scale the mask logits to the preprocessed image size during postprocessing (same for semantic and panoptic segmentation), even if no` target_sizes` is passed. I think it'd better to add an image processor for Mask2Former as its post-processing requires additional scaling.

What do you think @NielsRogge @sgugger?

**Comment by NielsRogge** [comment]
If postprocessing is different, then it indeed requires its own image processor class.

File: tests/models/maskformer/test_image_processing_maskformer.py [hunk 1]
```diff
@@ -598,7 +600,9 @@ def test_post_process_instance_segmentation(self):
             self.assertTrue("segments_info" in el)
             self.assertEqual(type(el["segments_info"]), list)
             self.assertEqual(len(el["segmentation"].shape), 3)
-            self.assertEqual(el["segmentation"].shape[1:], (384, 384))
+            self.assertEqual(
```
Comment (NielsRogge) [inline tests/models/maskformer/test_image_processing_maskformer.py:603]: Shouldn't we also test expected slices?

Comment (alaradirik) [reply tests/models/maskformer/test_image_processing_maskformer.py:603]: @NielsRogge I can add a test to the modeling test file since this is a dummy file but it'd great to merge this PR first.

Comment (NielsRogge) [reply tests/models/maskformer/test_image_processing_maskformer.py:603]: Ok!

**Review by sgugger — APPROVED** [review]
